Setting-Up-Ubuntu-For-Programming.md

```diff
@@ -0,0 +1,219 @@
+## Customising Ubuntu
+
+#### `.deb` Package Installer
+
+Install gdebi (`.deb` package installer) by typing `sudo apt-get install gdebi`.
+
+#### Removing Bloatware
+
+To remove all pre-installed bloatware because of privacy concerns or to keep your operating system minimal, checkout [this](https://gist.github.com/ansell/61313400e26cd42289f8) gist.
+
+#### Aliases
+
+You can create a temporary alias like this:
+```bash
+alias alias_name="command_to_run"
+```
+However, when you close your shell session, this alias will cease to exist.
+
+To create a permanent alias you will need to edit the `~/.bash_aliases` file. Once you've opened this file with your text editor of choice, add a line at the bottom of the document, similar to the example above.
+
+To find out more, DigitalOcean has a great tutorial that can be found [here](https://www.digitalocean.com/community/tutorials/an-introduction-to-useful-bash-aliases-and-functions).
+
+#### Unity Tweak Tool
+
+The Unity Tweak Tool provides users with tons of configuration options for tweaking the Unity Desktop.
+
+To install the Unity Tweak Tool type `sudo apt-get install unity-tweak-tool`, and to launch it, `unity-tweak-tool`.
+
+[Here](http://www.techrepublic.com/blog/linux-and-open-source/six-must-have-ubuntu-unity-tweaks/) is a list of the six must-have Ubuntu Unity Tweaks.
+
+## Jazzing up the Terminal
+
+Out of the box, the terminal shipped with Ubuntu is a bit bland. This section will let you become a power-user :muscle:.
+
+#### Terminator
+
+[Terminator](https://launchpad.net/terminator) lets you arrange multiple terminals in a grid-like interface.
+To install Terminator enter `sudo apt-get install terminator` into the terminal.
+
+Screenshot of Terminator:
+
+![Screenshot of Terminator](http://2.bp.blogspot.com/-DYWH3IZubRI/TrmFBPxmyEI/AAAAAAAAAAA/nBGoP3U3ae4/s1600/terminator-mad.png)
+
+#### Oh My ZSH!
+
+Prerequisites:
+- `git` should be installed
+
+To install `ZSH` and `Oh My ZSH!` use the following commands:
+
+```bash
+sudo apt-get install zsh && chsh -s $(which zsh)
+```
+
+> Note: you will need to log out and in again to use ZSH instead of bash as your default shell.
+
+```bash
+sh -c "$(curl -fsSL https://raw.githubusercontent.com/robbyrussell/oh-my-zsh/master/tools/install.sh)"
+```
+
+Once you restart your terminal, `Oh My ZSH!` should be installed.
+
+Checkout the [Official Documentation](https://github.com/robbyrussell/oh-my-zsh/wiki) to learn how to install plugins and themes.
+
+#### Advanced Tutorials
+
+- [The Command Line Crash Course](http://cli.learncodethehardway.org/book/)
+- [The Art of Command Line](https://github.com/jlevy/the-art-of-command-line)
+- [Learn Enough Command Line to Be Dangerous](https://www.learnenough.com/command-line-tutorial)
+
+## Installing DevTools and modern web browsers
+
+#### Git
+```bash
+sudo apt-get install git
+```
+
+#### Node JS
+```bash
+curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.31.0/install.sh | bash
+nvm install 5.0
```

Comment:
6.0 is the latest version, you should mention that `nvm install stable` will install the current latest version. That command you put will specify a particular version.